overlord/snapstate/handlers.go

```diff
@@ -460,6 +468,52 @@ func (m *SnapManager) cleanupCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *SnapManager) createSnapContext(st *state.State, snapName string) error {
+	contextID := strutil.MakeRandomString(40)
```

Comment:
Right. The discrepancy arose after I switched from random bytes + base64 encoding to MakeRandomString and forgot to update it in snap-confine, and the code and tests are only making sure we don't exceed 44 characters.
Added a spread test to check we have the exact length.